x-pack/test/ui_capabilities/common/types.ts

```diff
@@ -22,7 +22,7 @@ export interface RoleKibanaPrivilege {
 export interface CustomRoleSpecification {
   name: string;
   elasticsearch?: {
-    cluster: string[];
+    cluster?: string[];
```

Comment:
@elastic/kibana-security I'm not sure how this used to work, but it seems this property was effectively optional since many role specifications don't specify them:

```
x-pack/test/ui_capabilities/security_and_spaces/scenarios.ts:12:3 - error TS2326: Types of property 'elasticsearch' are incompatible.
  Property 'cluster' is missing in type '{ indices: { names: string[]; privileges: string[]; }[]; }' but required in type '{ cluster: string[]; indices: CustomRoleSpecificationElasticsearchIndices[]; }'.

12   role: {
      ~~~~

  x-pack/test/ui_capabilities/common/types.ts:25:5
    25     cluster: string[];
            ~~~~~~~
    'cluster' is declared here.
  x-pack/test/ui_capabilities/common/types.ts:48:3
    48   role?: ReservedRoleSpecification | CustomRoleSpecification;
          ~~~~
    The expected type comes from property 'role' which is declared here on type 'User'

x-pack/test/ui_capabilities/security_and_spaces/scenarios.ts:38:3 - error TS2326: Types of property 'elasticsearch' are incompatible.
  Property 'cluster' is missing in type '{ indices: { names: string[]; privileges: string[]; }[]; }' but required in type '{ cluster: string[]; indices: CustomRoleSpecificationElasticsearchIndices[]; }'.

38   role: {
      ~~~~

  x-pack/test/ui_capabilities/common/types.ts:25:5
    25     cluster: string[];
            ~~~~~~~
    'cluster' is declared here.
  x-pack/test/ui_capabilities/common/types.ts:48:3
    48   role?: ReservedRoleSpecification | CustomRoleSpecification;
          ~~~~
    The expected type comes from property 'role' which is declared here on type 'User'

x-pack/test/ui_capabilities/security_and_spaces/scenarios.ts:55:3 - error TS2326: Types of property 'elasticsearch' are incompatible.
  Property 'cluster' is missing in type '{ indices: { names: string[]; privileges: string[]; }[]; }' but required in type '{ cluster: string[]; indices: CustomRoleSpecificationElasticsearchIndices[]; }'.

55   role: {
      ~~~~

  x-pack/test/ui_capabilities/common/types.ts:25:5
    25     cluster: string[];
            ~~~~~~~
    'cluster' is declared here.
  x-pack/test/ui_capabilities/common/types.ts:48:3
    48   role?: ReservedRoleSpecification | CustomRoleSpecification;
          ~~~~
    The expected type comes from property 'role' which is declared here on type 'User'
```

Reply:
This seems like a reasonable change. Can we update the sibling `indices` property to be optional too, to be consistent?